### PR TITLE
Create app

### PR DESCRIPTION
**Pull request by Xodoc**

### CODE REVIEW

**Review by alesyalis — APPROVED** [review]

**Review by ihhyy — APPROVED** [review]

**Review by denja2 — CHANGES_REQUESTED** [review]

File: ConsoleApp/BL/Services/WeatherService.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        {
+            _unitOfWork = unitOfWork;
+            _validator = validator;
+        }
+
+        public async Task<WeatherNowDTO> GetWeatherAsync(string cityName)
+        {
+            var weather = await _unitOfWork.WeatherRepository.GetWeatherAsync(cityName);
+
+            _validator.ValidateIfEntityExist(weather);
```
[the remark this answers is not in the document]
Comment (Xodoc) [reply ConsoleApp/BL/Services/WeatherService.cs:24]: done

File: ConsoleApp/BL/Services/WeatherService.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+            weather.weather[0].description = GetWeaatherDescription(weather);
+
+            return Mapping(weather);
+        }
+
+        private string GetWeaatherDescription(Root weather)
```
[the remark this answers is not in the document]
Comment (Xodoc) [reply ConsoleApp/BL/Services/WeatherService.cs:31]: done

File: ConsoleApp/BL/Services/WeatherService.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            _validator = validator;
+        }
+
+        public async Task<WeatherNowDTO> GetWeatherAsync(string cityName)
+        {
+            var weather = await _unitOfWork.WeatherRepository.GetWeatherAsync(cityName);
+
+            _validator.ValidateIfEntityExist(weather);
+
+            weather.weather[0].description = GetWeaatherDescription(weather);
```
[the remark this answers is not in the document]
Comment (Xodoc) [reply ConsoleApp/BL/Services/WeatherService.cs:26]: done

File: ConsoleApp/BL/Services/WeatherService.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            if (weather.main.temp < 0)
+                return "Dress warmly.";
+            if (weather.main.temp >= 0 && weather.main.temp <= 20)
+                return "It's fresh.";
+            if (weather.main.temp >= 30 && weather.main.temp <= 30)
+                return "Good weather.";
+            if (weather.main.temp >= 30)
+                return "it's time to go to the beach.";
+
+            return weather.weather[0].description;
```
[the remark this answers is not in the document]
Comment (Xodoc) [reply ConsoleApp/BL/Services/WeatherService.cs:42]: done

File: ConsoleApp/BL/Services/WeatherService.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        }
+
+        private WeatherNowDTO Mapping(Root weather)
+        {
+            var weatherNowDTO = new WeatherNowDTO
```
[the remark this answers is not in the document]
Comment (Xodoc) [reply ConsoleApp/BL/Services/WeatherService.cs:47]: done

File: ConsoleApp/BL/Validators/WeatherValidator.cs [hunk 6]
```diff
@@ -0,0 +1,8 @@
+﻿using DAL.Entities;
+
+namespace BL.Validators
+{
+    public class WeatherValidator : Validator<Root>
```
[the remark this answers is not in the document]
Comment (Xodoc) [reply ConsoleApp/BL/Validators/WeatherValidator.cs:5]: done

File: ConsoleApp/Shared/Constants/RequestConstants.cs [hunk 7]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Shared.Constants
+{
+    public static class RequestConstants
+    {
+        public const string UNITS = "metric";
```
[the remark this answers is not in the document]
Comment (Xodoc) [reply ConsoleApp/Shared/Constants/RequestConstants.cs:5]: done

File: ConsoleApp/DAL/Repositories/WeatherRepository.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            _client = new HttpClient();
+            _configuration = configuration;
+        }
+
+        public async Task<Root> GetWeatherAsync(string cityName)
+        {
+            try
+            {
+                var weather = await _client.GetStringAsync($"{URL}{cityName}&lang={LANG}&units={UNITS}&appid={_configuration.APIKey}");
```
[the remark this answers is not in the document]
Comment (Xodoc) [reply ConsoleApp/DAL/Repositories/WeatherRepository.cs:28]: done

File: ConsoleApp/ConsoleApp/Program.cs [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+        {
+            NinjectModule serviceModule = new ServiceModule();
+
+            var kernel = new StandardKernel(serviceModule);
+            kernel.Load(Assembly.GetExecutingAssembly());
+            DependencyResolver.SetResolver(new NinjectDependencyResolver(kernel));
+
+            _weatherService = kernel.Get<IWeatherService>();
+            _configuration = kernel.Get<IConfiguration>();
+            _configuration.APIKey = ConfigurationManager.AppSettings["apiKey"];
```
[the remark this answers is not in the document]
Comment (Xodoc) [reply ConsoleApp/ConsoleApp/Program.cs:67]: done

File: ConsoleApp/DAL/Repositories/UnitOfWork.cs [hunk 10]
```diff
@@ -0,0 +1,36 @@
+﻿using DAL.Interfaces;
+using Shared.Config;
+using Shared.Interfaces;
+
+namespace DAL.Repositories
+{
+    public class UnitOfWork : IUnitOfWork
```
[the remark this answers is not in the document]
Comment (Xodoc) [reply ConsoleApp/DAL/Repositories/UnitOfWork.cs]: done

**Pull request merged**